pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/ClinicalOrderDao.java

```diff
@@ -0,0 +1,113 @@
+                                                          ClinicalOrderUseCase clinicalOrderUseCase) {
+        DDPInstanceDto ddpInstance = new DDPInstanceDao().getDDPInstanceByInstanceName(realm).orElseThrow();
+        HashMap<String, ClinicalOrderDto> map = new HashMap<>();
+        ArrayList<ClinicalOrderDto> noStatusOrders = new ArrayList<>();
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_GET_ALL_ORDERS_FOR_REALM)) {
+                stmt.setInt(1, ddpInstance.getDdpInstanceId());
+                try (ResultSet rs = stmt.executeQuery()) {
+                    while (rs.next()) {
+                        String key = rs.getString(DBConstants.COLLABORATOR_SAMPLE_ID) + "_" + rs.getString(DBConstants.MERCURY_ORDER_ID);
+                        if (!map.containsKey(key)) {
+                            String shortId = createShortIdFromCollaboratorSampleId(rs.getString(DBConstants.COLLABORATOR_SAMPLE_ID));
+                            ClinicalOrderDto clinicalOrderDto = new ClinicalOrderDto(shortId,
+                                    rs.getString(DBConstants.COLLABORATOR_SAMPLE_ID), rs.getString(DBConstants.MERCURY_ORDER_ID),
+                                    rs.getString(DBConstants.MERCURY_ORDER_STATUS), rs.getLong(DBConstants.MERCURY_ORDER_DATE),
+                                    rs.getLong(DBConstants.MERCURY_STATUS_DATE), rs.getString(DBConstants.MERCURY_STATUS_DETAIL),
+                                    rs.getString(DBConstants.MERCURY_SAMPLE_TYPE)
+                            );
+                            map.put(key, clinicalOrderDto);
+                            if (!COMPLETED_ORDER_STATUS.equals(clinicalOrderDto.getStatusDetail())) {
+                                noStatusOrders.add(clinicalOrderDto);
+                            }
+                        }
+                    }
+                } catch (Exception e) {
+                    dbVals.resultException = e;
+                }
+            } catch (SQLException ex) {
+                dbVals.resultException = ex;
+            }
+
+            return dbVals;
+        });
+
+        if (results.resultException != null) {
+            throw new RuntimeException("Error getting mercury orders for realm " + realm,
+                    results.resultException);
+        }
+        ArrayList<ClinicalOrderDto> array = new ArrayList(map.values());
+        log.info(String.format("Found %d orders for realm %s", array.size(), realm));
+        Collections.sort(array, new Comparator<ClinicalOrderDto>() {
+            @Override
+            public int compare(ClinicalOrderDto c1, ClinicalOrderDto c2) {
+                return (int) (c2.orderDate - c1.orderDate);
+            }
+        });
+        clinicalOrderUseCase.publishStatusActionMessage(array, projectId, topicId);
```

Comment:
not a good idea, `DAO` classes and their methods should have only one responsibility and it should be related only to CRUD operations, remove this operation from here and do it separately where you call `getOrdersForRealm`.